### ISSUE TITLE
Add git amend -i for a TUI

### ISSUE THREAD

**Issue by waylon-brown** [arxanas/git-branchless#1019]
### Description of the bug

For `git amend` the docs state that if I want to amend only _some_ changes, I should add those to the staging area first. 

I then expected there to be something similar to `git record -i` where I can choose with the TUI which changes to stage. This would be great so that I don't need to create a new commit each time I want to use the TUI to add only some changes.

**Comment by arxanas** [comment]
I agree that this would be useful, but I don't have any plans to work on it. You could try https://github.com/martinvonz/jj which *does* have a `jj amend -i` command.

**Comment by waylon-brown** [comment]
Out of curiosity, why is it this isn't something that's planned? Is it too much work to integrate the existing TUI with amend? The TUI is _really_ great - I think it'd make branchless a bit more intuitive to be able to use the TUI in other places since it's working with `record`.

**Comment by waylon-brown** [comment]
For ex. I can accomplish it w/ a couple commands, so I'm wondering if much of the internals could be reused here.

```
# git stage interactive
# Select changes in working directory to stage.
function gsi {
	# Using && \ to abort if existing TUI.
	# Commit changes to keep
	git record -i -m "Temp." && \
	# Move changes back into working directory, which has prev-commited changes staged.
	git reset --soft HEAD~
}

# git amend interactive
alias gai="gsi && git amend"
```

**Comment by arxanas** [comment]
> Out of curiosity, why is it this isn't something that's planned? Is it too much work to integrate the existing TUI with amend? The TUI is really great - I think it'd make branchless a bit more intuitive to be able to use the TUI in other places since it's working with record.

I just haven't had time to work on it. It should be reasonably straightforward if you want to work on it (but not trivial, because you have to synthesize the partial commit — you can see how it's done here:

https://github.com/arxanas/git-branchless/blob/ed35cdeccebb077af710ebcb342e4a032842f056/git-branchless-record/src/lib.rs#L304

).

> For ex. I can accomplish it w/ a couple commands, so I'm wondering if much of the internals could be reused here.

You would also want to support the `--merge` and `--reparent` options that already exist for `git amend`.

**Comment by WaylonBrown** [comment]
Thanks for the update - makes sense!

**Comment by ppwwyyxx** [comment]
Tried and found that `jj amend -i` by default includes untracked files. While `git record -i` does not. 

As a git user, I'm more comfortable with the behavior of `git record` where untracked files are ignored.

**Comment by martinvonz** [comment]
We have https://github.com/martinvonz/jj/issues/323 for that
